sdk/communication/communication-phone-numbers/src/utils/customPipelinePolicies.ts

```diff
@@ -0,0 +1,59 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { FullOperationResponse } from "@azure/core-client";
+import {
+  PipelinePolicy,
+  PipelineRequest,
+  PipelineResponse,
+  SendRequest,
+} from "@azure/core-rest-pipeline";
+
+/**
+ * Creates a `PipelinePolicy` that converts relative URL values in the `nextLink` property to absolute URLs.
+ *
+ * This is necessary because the Core V2 library does not support paging with relative links at time of writing.
+ *
+ * @param host - The base URL of the resource.
+ * @returns the `PipelinePolicy` that addresses the issue.
+ */
+export function createPhoneNumbersPagingPolicy(host: string): PipelinePolicy {
+  return {
+    name: "phoneNumbersPagingPolicy",
+    async sendRequest(request: PipelineRequest, next: SendRequest): Promise<PipelineResponse> {
+      const response: FullOperationResponse = await next(request);
+      const nextLink: string = response?.parsedBody?.nextLink;
+
+      if (nextLink && !nextLink.startsWith(host)) {
+        const absolutePath = host.endsWith("/") ? host + nextLink.substring(1) : host + nextLink;
```

Comment:
Here you are doing two thing at once, fixing the `nextLink` var and creating the absolute path, for me it would make more sense to split this two steps, so it is easier to identify the logic to create the final path, and as a bonus you adhere more to the DRY principle.

Reply:
Do you mean something like this (considering that `nextLink` is changed so it can be reassigned):
```suggestion
        nextLink =  host.endsWith("/") ? nextLink.substring(1) : nextLink;
        const absolutePath = host + nextLink;
```